tasks/configure_agent.yml

```diff
@@ -17,20 +17,30 @@
     - nrinfragent_os_name != 'windows'
     - ansible_python_version is version('3.0', '>=')
 
-- name: setup agent config *NIX
-  merge_yaml:
-    src: "{{ nrinfragent_config_path }}"
-    value: "{{ nrinfragent_config }}"
+- name: Ensure newrelic-infra.yml exists
+  file:
+    path: {{ nrinfragent_config_path }}
+    state: touch
+    mode: 0600
+  when: nrinfragent_os_name != 'windows'
+
+- name: Setup agent config *NIX
+  lineinfile:
+    'dest={{ nrinfragent_config_path }}
+    regexp="^{{ item.key }}: "
+    line="{{ item.key }}: {{ item.value }}"'
+  with_items: "{{ lookup('dict', nrinfragent_config, wantlist=True) }}"
+  no_log: "{{ nrinfragent_hide_config_values }}"
```

Comment:
@JakubKotkowiak this doesn't look like it merged properly.  The `lineinfile` is the old way.  We need to be using `merge_yaml`.

Reply:
Sure thing, I'll have to verify if merge_yaml doesn't have issues with vault first (this is the reason why I left the lineinfile)

Reply:
@JakubKotkowiak approving the changes thanks!  I _am_ curious though about your comment about vault.  Not sure what you mean by "vault"?